src/ralph/networks/filters.py

```diff
@@ -45,3 +50,26 @@ def queryset(self, request, queryset):
 
             queryset = queryset.filter(min_ip__gte=min_ip, max_ip__lte=max_ip)
         return queryset
+
+
+class NetworkClassFilter(ChoicesListFilter):
+    _choices_list = [
+        ('private', _('Private')),
+        ('public', _('Public')),
+    ]
+
+    def queryset(self, request, queryset):
+        if not self.value():
+            return queryset
+
+        filter_ = Q()
+        for private_cidr in PRIVATE_NETWORK_CIDRS:
+            network = ipaddress.ip_network(private_cidr)
+            min_ip = int(network.network_address)
+            max_ip = int(network.broadcast_address)
+            filter_ |= Q(min_ip__gte=min_ip, max_ip__lte=max_ip)
```

Comment:
you could construct this query (filters) only once, during application loading (or maybe during first call to this filter). There is no need to parse this with each request.